src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_verifier_index.rs

```diff
@@ -261,3 +315,131 @@ pub fn caml_pasta_fp_plonk_verifier_index_deep_copy(
+    };
+    use crate::srs::fp::CamlFpSrs;
+
+    fn comm() -> CamlPolyComm<CamlGVesta> {
+        let g: CamlGVesta = Vesta::prime_subgroup_generator().into();
+        CamlPolyComm {
+            shifted: Some(g),
+            unshifted: vec![g, g, g],
+        }
+    }
+
+    fn vec_comm(num: usize) -> Vec<CamlPolyComm<CamlGVesta>> {
+        (0..num).map(|_| comm()).collect()
+    }
+
+    // Helper function to generate a string representation of a value that
+    // derives the Debug trait.  This can be used to get a poor man's
+    // PartialEq/Eq-like trait for assertions when nothing else can be done
+    fn debug_string<G: std::fmt::Debug>(vi: G) -> String {
+        use std::fmt::Write;
+        let mut buf = String::new();
+        buf.write_fmt(format_args!("{:?}", vi))
+            .expect("a Debug implementation returned an error unexpectedly");
+        buf.shrink_to_fit();
+        buf
+    }
+
+    // Generate a verifier index value.
+    fn gen_caml_verifier_index() -> CamlPastaFpPlonkVerifierIndex {
+        CamlPlonkVerifierIndex {
+            domain: CamlPlonkDomain {
+                log_size_of_group: 2,
+                group_gen: Fp::one().into(),
+            },
+            max_poly_size: 0,
+            public: 0,
+            prev_challenges: 0,
+            srs: CamlFpSrs::new(SRS::create(0)),
+            evals: CamlPlonkVerificationEvals {
+                sigma_comm: vec_comm(PERMUTS),
+                coefficients_comm: vec_comm(COLUMNS),
+                generic_comm: comm(),
+                psm_comm: comm(),
+                complete_add_comm: comm(),
+                mul_comm: comm(),
+                emul_comm: comm(),
+                endomul_scalar_comm: comm(),
+                xor_comm: Some(comm()),
+                range_check0_comm: Some(comm()),
+                range_check1_comm: Some(comm()),
+                foreign_field_add_comm: Some(comm()),
+                foreign_field_mul_comm: Some(comm()),
+                rot_comm: Some(comm()),
+                lookup_gate_comm: Some(comm()),
+                runtime_tables_comm: Some(comm()),
+            },
+            shifts: (0..PERMUTS).map(|_| Fp::one().into()).collect(),
+            lookup_index: None,
+        }
+    }
+
+    #[test]
+    fn back_and_forth() -> () {
+        let dummy: CamlPastaFpPlonkVerifierIndex = gen_caml_verifier_index();
+
+        let vi = VerifierIndex::<Vesta>::from(dummy);
+        let caml_vi = CamlPastaFpPlonkVerifierIndex::from(vi.clone());
+        let vi2 = VerifierIndex::<Vesta>::from(caml_vi);
+
+        //////////////////////////////////////////////////////////////////////////////
+        // The full equality assertion is divided up into assertions on fields.     //
+        // This helps in localizing errors when they fail.                          //
+        // The main issue is that one could easily forget a record field.           //
```

Comment:
TODO-later: if you use destructuring assignment (that is `let A {foo, bar, baz} = x`) then the compiler will warn on missing fields.

Reply:
Very nice idea and exactly what I wanted. I have implemented it.